pkg/compiler/transformers/k8s/node.go

```diff
@@ -221,6 +221,18 @@ func buildTasks(tasks []*core.CompiledTask, errs errors.CompileErrors) map[commo
 				errs.Collect(errors.NewValueCollisionError(taskID, "Id", taskID))
 			}
 
+			// We don't want Annotation data available at task runtime for performance.
+			if flyteTask.Template != nil &&
```

Comment:
i'm going to be honest this is pretty gross, but don't know the most semantic way of dereferencing nested pointers here given that the whole struct is received as a big chunk from admin